File: leetcode/66_plus_one.py
from typing import List


class Solution:

    def plusOne(self, digits: List[int]) -> List[int]:
        N = len(digits)
        for i in reversed(range(N)):
            digit = digits[i]
            if digit == 9:
                digits[i] = 0
            else:
                digits[i] += 1
                return digits
        digits[0] = 1
        digits.append(0)
        return digits

File: leetcode/java/66PlusOne.java
class Solution {
  public int[] plusOne(int[] digits) {
    int digitsCount = digits.length;
    for (int i = digitsCount - 1; i >= 0; --i) {
      if (digits[i] == 9) {
        digits[i] = 0;
      } else {
        digits[i]++;
        return digits;
      }
    }
    digits = new int[digitsCount + 1];
    digits[0] = 1;
    return digits;
  }
}
